src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -541,3 +549,68 @@ updateExtConfigArray(Archive *fout, char ***extconfigarray, int nconfigitems)
 	PQclear(res);
 	resetPQExpBuffer(query);
 }
+
+/*
+ * fixCopyCommand:
+ * Fixes column list in a COPY command.
+ * isFrom decides whether we are copying FROM or TO.
+ */
+void
+fixCopyCommand(Archive *fout, PQExpBuffer copyBuf, TableInfo *tbinfo, bool isFrom)
+{
+	PQExpBuffer q;
+	bool	    needComma;
+	bool	    found;
+	int 	    i;
+
+	/* Proceed only if we are dealing with sys.babelfish_authid_login_ext table. */
+	if (!isBabelfishDatabase(fout) || tbinfo->dobj.namespace == NULL ||
+		pg_strcasecmp(tbinfo->dobj.namespace->dobj.name, "sys") != 0 ||
+		pg_strcasecmp(tbinfo->dobj.name, "babelfish_authid_login_ext") != 0)
+		return;
+
+	q = createPQExpBuffer();
+	needComma = false;
+	for (i = 0; i < tbinfo->numatts; i++)
+	{
+		if (tbinfo->attisdropped[i])
+			continue;
+		if (tbinfo->attgenerated[i])
+			continue;
+		if (needComma)
+			appendPQExpBufferStr(q, ", ");
+		appendPQExpBufferStr(q, fmtId(tbinfo->attnames[i]));
+		if (pg_strcasecmp(tbinfo->attnames[i], "orig_loginname") == 0)
+			found = true;
+		needComma = true;
+	}
+
+	/* orig_loginname column not found */
+	if (!found)
+	{
+		resetPQExpBuffer(copyBuf);
+		appendPQExpBufferStr(q, ", ");
+		/* Append orig_loginname column at the end in COPY FROM command */
```

Comment:
is it safe to assume that orig_loginname is last column? what if a new column is added in the future?